pkg/tsdb/cloudwatch/cloudwatch_query_test.go

```diff
@@ -9,27 +9,107 @@ import (
 )
 
 func TestCloudWatchQuery(t *testing.T) {
-	t.Run("Deeplink is not generated for MetricQueryTypeQuery", func(t *testing.T) {
-		startTime := time.Now()
-		endTime := startTime.Add(2 * time.Hour)
-		query := &cloudWatchQuery{
-			RefId:      "A",
-			Region:     "us-east-1",
-			Expression: "",
-			Statistic:  "Average",
-			Period:     300,
-			Id:         "id1",
-			MatchExact: true,
-			Dimensions: map[string][]string{
-				"InstanceId": {"i-12345678"},
+	t.Run("Deeplink", func(t *testing.T) {
+		t.Run("is not generated for MetricQueryTypeQuery", func(t *testing.T) {
+			startTime := time.Now()
+			endTime := startTime.Add(2 * time.Hour)
+			query := &cloudWatchQuery{
+				RefId:      "A",
+				Region:     "us-east-1",
+				Expression: "",
+				Statistic:  "Average",
+				Period:     300,
+				Id:         "id1",
+				MatchExact: true,
+				Dimensions: map[string][]string{
+					"InstanceId": {"i-12345678"},
+				},
+				MetricQueryType:  MetricQueryTypeQuery,
+				MetricEditorMode: MetricEditorModeBuilder,
+			}
+
+			deepLink, err := query.buildDeepLink(startTime, endTime, false)
+			require.NoError(t, err)
+			assert.Empty(t, deepLink)
+		})
+
+		testCases := map[string]struct {
+			linkContainsSubstring string
+			query                 *cloudWatchQuery
+			featureEnabled        bool
+		}{"includes label in case dynamic label is enabled and its a metric stat query": {
+			query: &cloudWatchQuery{
+				RefId:      "A",
+				Region:     "us-east-1",
+				Expression: "",
+				Statistic:  "Average",
+				Period:     300,
+				Id:         "id1",
+				MatchExact: true,
+				Label:      "${PROP('Namespace')}",
+				Dimensions: map[string][]string{
+					"InstanceId": {"i-12345678"},
+				},
+				MetricQueryType:  MetricQueryTypeSearch,
+				MetricEditorMode: MetricEditorModeBuilder,
+			},
+			featureEnabled: true,
+			// decoded: label: ${PROP('Namespace')}
+			linkContainsSubstring: "label%22%3A%22%24%7BPROP%28%27Namespace%27%29%7D%22%7D%5D%5D%7D",
+		},
+			"does not include label in case dynamic label is diabled": {
+				query: &cloudWatchQuery{
+					RefId:      "A",
+					Region:     "us-east-1",
+					Expression: "",
+					Statistic:  "Average",
+					Period:     300,
+					Id:         "id1",
+					MatchExact: true,
+					Label:      "${PROP('Namespace')}",
+					Dimensions: map[string][]string{
+						"InstanceId": {"i-12345678"},
+					},
+					MetricQueryType:  MetricQueryTypeSearch,
+					MetricEditorMode: MetricEditorModeBuilder,
+				},
+				featureEnabled: true,
+				// decoded: label: ${PROP('Namespace')}
+				linkContainsSubstring: "label%22%3A%22%24%7BPROP%28%27Namespace%27%29%7D%22%7D%5D%5D%7D",
+			},
+			"includes label in case dynamic label is enabled and its a math expression query": {
```

Comment:
I think this is exactly the same test as the first test case

Reply:
Yeah thanks for finding this! Actually table tests were not suitable here so did some refactoring here.